Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -302,7 +317,49 @@ public enum CommandLineArgumentsResult {
       }
 
       string programName = dafnyFileNames.Count == 1 ? dafnyFileNames[0] : "the_program";
-      string err = Dafny.Main.ParseCheck(dafnyFiles, programName, reporter, out var dafnyProgram);
+      Program dafnyProgram;
+      string err;
+      if (Action == DriverAction.Format) {
+        if (dafnyFiles.Count == 0) {
+          // Let's list all the dafny files recursively in the working directory
+          dafnyFiles = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.dfy", SearchOption.AllDirectories)
+            .Select(name => new DafnyFile(name)).ToList();
+        }
+
+        var errors = 0;
+        var unchanged = 0;
+        foreach (var dafnyFile in dafnyFiles) {
+          // Might not be totally optimized but let's do that for now
+          err = Dafny.Main.Parse(new List<DafnyFile> { dafnyFile }, programName, reporter, out dafnyProgram);
+          if (err != null) {
+            exitValue = ExitValue.DAFNY_ERROR;
+            Console.Error.WriteLine(err);
+            errors += 1;
+          } else {
+            var firstToken = dafnyProgram.GetFirstTopLevelToken();
+            if (firstToken != null) {
+              var result = Formatting.__default.printSourceReindent(firstToken,
+                IndentationFormatter.ForProgram(dafnyProgram));
+              if (DafnyOptions.O.PrintFile == "-") {
+                Console.Out.Write(result);
+              } else {
+                WriteFile(dafnyFile.FilePath, result);
+              }
+            } else {
+              Console.Error.WriteLine("Could not find text there");
```

Comment:
Am I understanding correctly that this branch occurs when a file successfully parses and contains no tokens? If that's the case then should we really count that as an error?

Reply:
First, it's unexpected that the formatter will find an empty file. Second. I had seen cases (not anymore) when the formatter wrongly thought that there were no content when there was. Hence, I would like to report them as well, just in case.